pyogrio/tests/test_arrow.py

```diff
@@ -161,6 +161,35 @@ def test_read_arrow_vsi(naturalearth_lowres_vsi):
     assert len(table) == 177
 
 
+@requires_arrow_write_api
+@pytest.mark.filterwarnings("ignore:A geometry of type POLYGON is inserted")
+@pytest.mark.filterwarnings("ignore:File /vsimem:RuntimeWarning")
+@pytest.mark.parametrize("driver,ext", [("GeoJSON", "geojson"), ("GPKG", "gpkg")])
+def test_read_from_bytes(tmp_path, naturalearth_lowres, driver, ext):
+    meta, table = read_arrow(naturalearth_lowres)
+    meta["geometry_type"] = "MultiPolygon"
+
+    filename = tmp_path / f"test.{ext}"
+
+    write_arrow(
+        table,
+        filename,
+        driver=driver,
+        layer="test",
+        crs=meta["crs"],
+        geometry_type=meta["geometry_type"],
+        geometry_name=meta["geometry_name"] or "wkb_geometry",
+    )
+
+    with open(filename, "rb") as f:
+        buffer = f.read()
```

Comment:
Is there a need to write this first, instead of directly reading the buffer from `naturalearth_lowres` ?

Reply:
(and then this read test also doesn't require `requires_arrow_write_api`)

Reply:
Or, could also use the new fixtures `geojson_bytes` and `geojson_filelike` here?